opentracing-mock/src/main/java/io/opentracing/mock/MockTracer.java

```diff
@@ -112,6 +119,82 @@ protected void onSpanFinished(MockSpan mockSpan) {
             }
         };
 
+        Propagator BINARY = new Propagator() {
+            static final int BUFFER_SIZE = 128;
+
+            @Override
+            public <C> void inject(MockSpan.MockContext ctx, Format<C> format, C carrier) {
+                if (carrier instanceof Binary) {
+                    Binary binary = (Binary) carrier;
+                    ByteArrayOutputStream stream = new ByteArrayOutputStream();
+                    ObjectOutputStream objStream = null;
+                    try {
+                        objStream = new ObjectOutputStream(stream);
+                        objStream.writeLong(ctx.spanId());
+                        objStream.writeLong(ctx.traceId());
+
+                        for (Map.Entry<String, String> entry : ctx.baggageItems()) {
+                            objStream.writeUTF(entry.getKey());
+                            objStream.writeUTF(entry.getValue());
+                        }
+                        objStream.flush(); // *need* to flush ObjectOutputStream.
+
+                        byte[] buff = stream.toByteArray();
+                        binary.setInjectBufferLength(buff.length);
+                        binary.injectBuffer().put(buff);
+
+                    } catch (IOException e) {
+                        throw new RuntimeException("Corrupted state");
+                    } finally {
+                        if (objStream != null) {
+                            try { objStream.close(); } catch (Exception e2) {}
+                        }
+                    }
+                } else {
+                    throw new IllegalArgumentException("Unknown carrier");
+                }
+            }
+
+            @Override
+            public <C> MockSpan.MockContext extract(Format<C> format, C carrier) {
+                Long traceId = null;
+                Long spanId = null;
+                Map<String, String> baggage = new HashMap<>();
+
+                if (carrier instanceof Binary) {
+                    Binary binary = (Binary) carrier;
+                    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+                    ObjectInputStream objStream = null;
+                    try {
+                        ByteBuffer extractBuff = binary.extractBuffer();
+                        byte[] buff = new byte[extractBuff.remaining()];
+                        extractBuff.get(buff);
+
+                        objStream = new ObjectInputStream(new ByteArrayInputStream(buff));
+                        spanId = objStream.readLong();
+                        traceId = objStream.readLong();
+
+                        while (objStream.available() > 0) {
+                            baggage.put(objStream.readUTF(), objStream.readUTF());
+                        }
+                    } catch (IOException e) {
```

Comment:
rethrow as wrapped?

Reply:
So I'm wondering about this one - the `TEXT_MAP` format returns nothing (`null`) if the context is malformed (i.e. `trace_id` but no `span_id`, or nothing at all, etc). Should we re-throw the `Exception` anyway here, though?

Reply:
I think for mock tracer it's better to throw on malformed carrier. Null is only when http header is missing (which doesn't apply to binary codecs).